docs/serving/samples/cloudevents/cloudevents-spring/README.md

```diff
@@ -0,0 +1,139 @@
+supports running in two modes:
+
+1. The default mode has the app reply to your input events with the output
+   event, which is simplest for demonstrating things working in isolation, but
+   is also the model for working for the Knative Eventing `Broker` concept. The
+   input event is modified assigning a new source and type attribute.
+
+2. `K_SINK` mode has the app send events to the destination encoded in
+   `$K_SINK`, which is useful to demonstrate how folks can synthesize events to
+   send to a Service or Broker when not initiated by a Broker invocation (e.g.
+   implementing an event source). The input event is modified assigning a new
+   source and type attribute.
+
+The application will use `$K_SINK`-mode whenever the environment variable is
+specified.
+
+Follow the steps below to create the sample code and then deploy the app to your
+cluster. You can also download a working copy of the sample, by running the
+following commands:
+
+```shell
+git clone -b "{{< branch >}}" https://github.com/knative/docs knative-docs
+cd knative-docs/docs/serving/samples/cloudevents/cloudevents-spring
+```
+
+## Before you begin
+
+- A Kubernetes cluster with Knative installed and DNS configured. Follow the
+  [installation instructions](../../../../install/README.md) if you need to
+  create one.
+- [Docker](https://www.docker.com) installed and running on your local machine,
+  and a Docker Hub account configured (we'll use it for a container registry).
+
+## Build and deploy the sample
+
+To build the image, run:
+
+```shell
+mvn compile jib:build -Dimage=<image_name>
+```
+
+To deploy the Knative Service, look in the `service.yaml` and replace `<image>`
+with the deployed image name. Then run:
+
+```shell
+kubectl apply -f service.yaml
+```
+
+## Testing the sample
+
+Get the URL for your Service with:
+
+```shell
+$ kubectl get ksvc
+NAME                URL                                            LATESTCREATED             LATESTREADY               READY   REASON
+cloudevents-spring   http://cloudevents-java.xip.io                 cloudevents-spring-86h28   cloudevents-spring-86h28   True
+```
+
+Then send a CloudEvent to it with:
+
+```shell
+$ curl -v \
+    -H "content-type: application/json"  \
+    -H "ce-specversion: 1.0"  \
+    -H "ce-source: http://curl-command"  \
+    -H "ce-type: curl.demo"  \
+    -H "ce-id: 123-abc"  \
+    -d '{"name":"Dave"}' \
+    http://cloudevents-java.xip.io
+```
+
+You can also use the "structured" message format from Cloud Events using a more specific content type:
+
+```shell
+$ curl -v \
+    -H "content-type: application/cloudevents+json"  \
+    -d '{
+      "specversion": "1.0",
+      "source": "http://curl-command",
+      "type": "curl.demo",
+      "id": "123-abc",
+      "data": {"name":"Dave"}
+    }' \
+    http://cloudevents-java.xip.io
+```
+
+N.B. the response from the server is "binary", not "structured", so you get the same response as if you sent a binary message..
```

Comment:
Maybe expand what N.B. stands for, I'm not sure of what it means in the sentence.

Reply:
N.B. -> "Note". I'll put in another commit and then squash them all down.